### PR TITLE
Delay colorpick eyedrop preview using a timer

### LINKED ISSUE #25358: ColorPicker's eyedrop takes ages to pick a color
**Godot version:**
 1253b9c

**OS/device including version:**
Manjaro GNU/Linux - Intel i5, Haswell

**Issue description:**
Both in-game and in the editor, when trying to pick a color via the eyedrop, everything will start to freeze until much later, when the color will finally be picked.

### PR DESCRIPTION
**Pull request by giarve**
This avoids recalculating the color of the selected pixel every time the mouse is moved.

Fixes #25358.

### CODE REVIEW

File: scene/gui/color_picker.cpp
Comment (Calinou) [inline scene/gui/color_picker.cpp:864]: Add a comment to explain, as this isn't obvious at a glance:

```suggestion
	// Debounce color picker eyedrop preview to prevent the colorpicker from taking a long time to pick a color.
	mouse_not_moving_timer = memnew(Timer);
```

**Review by Calinou — CHANGES_REQUESTED** [review]
Should be good to merge after applying the suggestion and squashing commits.

File: scene/gui/color_picker.cpp
Comment (KoBeWi) [inline scene/gui/color_picker.cpp:679]: I'd move the event `is_valid()` check here. There's no point in starting the timer if the event is invalid. Also with the current code, `get_global_position()` might be called on invalid event.

**Review by KoBeWi — CHANGES_REQUESTED** [review]

**Comment by KoBeWi** [comment]
This is impossible to test due to #46334, so probably won't be merged until that is resolved (unless we merge without testing...).

**Pull request closed without merging**
